### PR TITLE
Add tf.regex_match for regex match support

### LINKED ISSUE #18264: tensorflow new feature request: tf.regex_match
The new tf.regex_replace feature in tensorflow 1.7.0 works fine for modifying data and returning tf.string. However to filter a dataset, a separate method is needed in order to return tf.bool for use in tf.dataset.filter

I would like to be able to filter blank lines, or lines with only whitespace from a dataset, like:
dataset = dataset.filter(lambda line: tf.regex_match(line, "^\s*$"))

Currently, I can do this using tf.py_func, see details: https://stackoverflow.com/questions/49654187/tensorflow-removing-blank-lines-in-dataset

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue raised in #18264.

Currently tf.regex_replace has already been supported though there was no regex match support.
This fix adds the tf.regex_match support in a similiar pattern as tf.regex_replace.

This fix fixes #18264.

Note: this PR is a preliminary implementation which support full match of the regex. Additional variations could be added or in follow up PRs if needed.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/core/api_def/base_api/api_def_RegexMatch.pbtxt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+The regular expression to match the input.
+END
+  }
+  out_arg {
+    name: "output"
+    description: <<END
+The same tensor shape as `input`.
+END
+  }
+  summary: "Check if the input matches the regex pattern."
```
Comment (rmlarsen) [inline tensorflow/core/api_def/base_api/api_def_RegexMatch.pbtxt]: Can you expand the documentation to describe the shapes and nature of the output. As I read it, the op takes a string tensor of any shape, and a scalar string tensor containing a single regex pattern. It returns a boolean tensor that of the same shape as the input, with values True for those inputs that match the pattern. Perhaps say that this expect regular expressions in the default format used by RE2 and give a link to the description of this.

**Review by rmlarsen — CHANGES_REQUESTED** [review]

**Review by asimshankar — COMMENTED** [review]
(From API review)
Thanks for the PR! 

- The op should be `RegexFullMatch` 
- The Python function should be `regex_full_match` (The "full" more clearly calls out that this is different from Python's `re.match`)
- Let's put this in the `tf.strings` namespace

File: tensorflow/core/api_def/base_api/api_def_RegexMatch.pbtxt (hunk 1 above)
Comment (yongtang) [reply tensorflow/core/api_def/base_api/api_def_RegexMatch.pbtxt]: Thanks @rmlarsen. The doc has been updated.

**Comment by yongtang** [comment]
Thanks @asimshankar @rmlarsen for the review. The PR has been updated with op exposed as `tf.strings.regex_full_match`. Please take a look and let me know if there are any issues.

**Review by rmlarsen — APPROVED** [review]

**Comment by rmlarsen** [comment]
@asimshankar handing to you for API approval.

**Comment by asimshankar** [comment]
Good for API addition. Thanks!

**Comment by rmlarsen** [comment]
Thanks!

**Pull request merged**